### PR TITLE
add `type` field to nats sink

### LINKED ISSUE #2103: Bug: miss "type" field in nats sink connector
### Where's the bug?

https://docs.risingwave.com/docs/current/sink-to-nats/

### Describe the bug

e.g. 
 I noticed that there is a error in this screenshot. 
This is incorrect because: The code require the input `type` filed. In previous time, I remember we have doc it. But now it seems we remove it in the doc sometime. I don't know why or it is an error. 
 I think you folks should add another filed named `type` and require input value is `append-only` 

For example, some user report when use the following cmd:
```
FROM test_nats_source
WITH (
    connector='nats',
    server_url = 'nats.default.svc.cluster.local:4222',
    subject = 'nats.rw_test'
    ...
);

```
It will meet error:
```
[2024-04-25 12:44:18] [XX000] ERROR: Failed to execute the statement
[2024-04-25 12:44:18] Caused by these errors (recent errors listed first):
[2024-04-25 12:44:18] 1: gRPC request to meta service failed: Internal error
[2024-04-25 12:44:18] 2: Sink error
[2024-04-25 12:44:18] 3: config error
[2024-04-25 12:44:18] 4: missing field `type`
```

 If you take a look at [this reference] you will agree with me. 
 Cheers.
image

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - Add a field `type` to nats sink of ver 1.9, 1.8, 1.7, 1.6

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/2103#event-12651841057

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Review by yufansong — APPROVED** [review]
LGTM

**Review by KeXiangWang — APPROVED** [review]
LGTM Thanks!

File: docs/guides/sink-to-nats.md [hunk 1]
```diff
@@ -72,3 +73,4 @@ According to the [NATS documentation](https://docs.nats.io/running-a-nats-servic
 |`connect_mode`|Required. Authentication mode for the connection. Allowed values: `plain`: No authentication; `user_and_password`: Use user name and password for authentication. For this option, `username` and `password` must be specified; `credential`: Use JSON Web Token (JWT) and NKeys for authentication. For this option, `jwt` and `nkey` must be specified.  |
 |`jwt` and `nkey`|JWT and NKEY for authentication. For details, see [JWT](https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_intro/jwt) and [NKeys](https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_intro/nkey_auth).|
 |`username` and `password`| Conditional. The client user name and pasword. Required when `connect_mode` is `user_and_password`.|
+|`type`|Required. Sink data type. Its value should be append-only.|
```
Comment (emile-00) [inline docs/guides/sink-to-nats.md]:
```suggestion
|`type`|Required. Sink data type. Its value should be `append-only`.|
```

File: versioned_docs/version-1.6/guides/sink-to-nats.md [hunk 2]
```diff
@@ -73,3 +74,4 @@ According to the [NATS documentation](https://docs.nats.io/running-a-nats-servic
 |`connect_mode`|Required. Authentication mode for the connection. Allowed values: `plain`: No authentication; `user_and_password`: Use user name and password for authentication. For this option, `username` and `password` must be specified; `credential`: Use JSON Web Token (JWT) and NKeys for authentication. For this option, `jwt` and `nkey` must be specified.  |
 |`jwt` and `nkey`|JWT and NKEY for authentication. For details, see [JWT](https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_intro/jwt) and [NKeys](https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_intro/nkey_auth).|
 |`username` and `password`| Conditional. The client user name and pasword. Required when `connect_mode` is `user_and_password`.|
+|`type`|Required. Sink data type. Its value should be append-only.|
```
Comment (emile-00) [inline versioned_docs/version-1.6/guides/sink-to-nats.md:77]:
```suggestion
|`type`|Required. Sink data type. Its value should be `append-only`.|
```

File: versioned_docs/version-1.7/guides/sink-to-nats.md [hunk 3]
```diff
@@ -72,3 +73,4 @@ According to the [NATS documentation](https://docs.nats.io/running-a-nats-servic
 |`connect_mode`|Required. Authentication mode for the connection. Allowed values: `plain`: No authentication; `user_and_password`: Use user name and password for authentication. For this option, `username` and `password` must be specified; `credential`: Use JSON Web Token (JWT) and NKeys for authentication. For this option, `jwt` and `nkey` must be specified.  |
 |`jwt` and `nkey`|JWT and NKEY for authentication. For details, see [JWT](https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_intro/jwt) and [NKeys](https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_intro/nkey_auth).|
 |`username` and `password`| Conditional. The client user name and pasword. Required when `connect_mode` is `user_and_password`.|
+|`type`|Required. Sink data type. Its value should be append-only.|
```
Comment (emile-00) [inline versioned_docs/version-1.7/guides/sink-to-nats.md:76]:
```suggestion
|`type`|Required. Sink data type. Its value should be `append-only`.|
```

File: versioned_docs/version-1.8/guides/sink-to-nats.md [hunk 4]
```diff
@@ -72,3 +73,4 @@ According to the [NATS documentation](https://docs.nats.io/running-a-nats-servic
 |`connect_mode`|Required. Authentication mode for the connection. Allowed values: `plain`: No authentication; `user_and_password`: Use user name and password for authentication. For this option, `username` and `password` must be specified; `credential`: Use JSON Web Token (JWT) and NKeys for authentication. For this option, `jwt` and `nkey` must be specified.  |
 |`jwt` and `nkey`|JWT and NKEY for authentication. For details, see [JWT](https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_intro/jwt) and [NKeys](https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_intro/nkey_auth).|
 |`username` and `password`| Conditional. The client user name and pasword. Required when `connect_mode` is `user_and_password`.|
+|`type`|Required. Sink data type. Its value should be append-only.|
```
Comment (emile-00) [inline versioned_docs/version-1.8/guides/sink-to-nats.md:76]:
```suggestion
|`type`|Required. Sink data type. Its value should be `append-only`.|
```

**Review by emile-00 — APPROVED** [review]
Added a minor suggestion, otherwise LGTM!

**Review by WanYixian — APPROVED** [review]

**Pull request merged**